library/ssl_tls13_client.c

```diff
@@ -559,6 +559,7 @@ static int ssl_tls13_parse_cookie_ext( mbedtls_ssl_context *ssl,
     MBEDTLS_SSL_CHK_BUF_READ_PTR( p, end, cookie_len );
     MBEDTLS_SSL_DEBUG_BUF( 3, "cookie extension", p, cookie_len );
 
+#if defined(MBEDTLS_SSL_PROTO_DTLS)
```

Comment:
Looking at this a bit more closely, this is not what we should do I think. We should rather in ssl_misc.h define `verify_cookie` and `verify_cookie_len` if `MBEDTLS_SSL_PROTO_DTLS` or `MBEDTLS_SSL_PROTO_TLS1_3` is defined, right ? I think also that we should not guard the current TLS 1.3 cookie code with MBEDTLS_SSL_COOKIE_C as it seems that this configuration option is rather for the server side.

Reply:
So. Does that mean we should support cookie in both dtls and tls 1.3? which we only extract verify_cookie and verify_cookie_len from  MBEDTLS_SSL_PROTO_DTLS resrictions and put them under #if defined(MBEDTLS_SSL_PROTO_DTLS)  || defined(MBEDTLS_SSL_PROTO_TLS1_3).    
About the MBEDTLS_SSL_COOKIE_C, should we change the name of the macro? or simply remove the guard?

Reply:
> Does that mean we should support cookie in both dtls and tls 1.3? which we only extract verify_cookie and verify_cookie_len from MBEDTLS_SSL_PROTO_DTLS resrictions and put them under #if defined(MBEDTLS_SSL_PROTO_DTLS) || defined(MBEDTLS_SSL_PROTO_TLS1_3).
Yes, this seems to be what I meant. 

> About the MBEDTLS_SSL_COOKIE_C, should we change the name of the macro? or simply remove the guard?
Just remove the guard.